apis/tempo/v1alpha1/microservices_webhook.go

```diff
@@ -192,6 +211,23 @@ func (v *validator) validateStorage(ctx context.Context, tempo *Microservices) f
 	return v.validateStorageSecret(tempo, storageSecret)
 }
 
+func (v *validator) validateReplicationFactor(tempo *Microservices) field.ErrorList {
```

Comment:
How about logging something in the operator when the replication factor is not valid?

Reply:
Afaics the user who edits the CR gets immediate feedback if the validation fails (`field.ErrorList`) and is not allowed to save the CR, so it's not possible for the user to overlook any validation error.